### DISCUSSION TITLE
danielgtaylor/jpeg-archive#jpeg-recompress support

### DISCUSSION THREAD

**Discussion by matteotrubini** [wintercms/winter#146] [Ideas]
https://github.com/danielgtaylor/jpeg-archive#jpeg-recompress

It could be useful as optional jpeg optimizer as it NOT require to define image quality,  but it rely on some [Image Comparison Metrics](https://github.com/danielgtaylor/jpeg-archive#image-comparison-metrics) to keep perceived visual quality.

As this task can be too long to be executed synchronously on page load, it could be queued or executed only via console, like any other cache warm-up.

**Comment by bennothommo** [comment]
@matteotrubini I could see this being done as part of a plugin, but it would not be suitable for core functionality, as we cannot assume that people have the ability to install CLI programs on their hosting platform.

**Reply by matteotrubini** [reply]
You are right, I think so, too.
Do you think it could be a plugin maintained by @wintercms team or to be a 3rd party one?

**Reply by LukeTowers** [reply]
Probably 3rd party, unless one of the core team builds the plugin and it's something that's generally applicable to a large number of users, we're not likely to want to support anything over and above the massive code base we already support.

**Reply by matteotrubini** [reply]
That's fully understandable :smile:

**Comment by Flynsarmy** [comment]
I like the idea of auto-optimizing images on or after upload. Rather than just jpegs, I found [this package](https://github.com/psliwa/image-optimizer) and sent a couple of pull requests to improve it a bit.

Are there hooks for successfully uploading an image, thumbnails created, image resized etc? I could probably whip up a plugin to handle the resizing.

**Comment by mjauvin** [comment]
ref. https://wintercms.com/docs/events/event/media.file.upload
ref. https://wintercms.com/docs/events/event/system.resizer.afterResize

**Comment by Flynsarmy** [comment]
First version of Image Optimizer plugin is complete. You can download it here https://github.com/Flynsarmy/wn-imageoptimizer-plugin
